lib/fluent/plugin/bigquery/writer.rb

```diff
@@ -317,6 +303,42 @@ def source_format
           "NEWLINE_DELIMITED_JSON"
         end
       end
+
+      def time_partitioning
+        return @time_partitioning if instance_variable_defined?(:@time_partitioning)
+
+        if @options[:time_partitioning_type]
+          @time_partitioning = {
+            type: @options[:time_partitioning_type].to_s.upcase,
+            field: @options[:time_partitioning_field] ? @options[:time_partitioning_field].to_s : nil,
+            expiration_ms: @options[:time_partitioning_expiration] ? @options[:time_partitioning_expiration] * 1000 : nil
+          }.compact
```

Comment:
I've fixed the code to use `Hash#reject` https://github.com/kaizenplatform/fluent-plugin-bigquery/pull/167/commits/06074eb4ca6057c3eb4656fa95f4cc3364ebd288.
In addition, I've changed some tests so that [they fail](https://travis-ci.org/kaizenplatform/fluent-plugin-bigquery/jobs/426601706) if we use `Hash#compact`. https://github.com/kaizenplatform/fluent-plugin-bigquery/pull/167/commits/c4d12f0268a16b6060846f14852802b664dddc09.